### PR TITLE
Fix Selection Tool in New Input System

### LINKED ISSUE #2539: Regression: Selection Tool not working with Styuls Button 2
**Affects versions :**
 - OS: Arch
 - (Linux only) Desktop environment: X11
 - Which version of libgtk do you use: libgtk: 3.24.23 
 - Version of Xournal++: Xournal++ 1.1.0+dev 
 - Installation method: build from 

**Describe the bug**
Lasso tool does no longer work when activated via Stylus Button 2:
image

**To Reproduce**
1. Set Stylus Button 2 to Select Region
2. Save Settings
3. Try to use Button 2 from Stylus

**Expected behavior**
Selected Region is shown while drawing and contained strokes are selected after completion.

**Screenshots of Problem**
No selected region is shown and nothing can be selected.
Peek 2020-12-13 17-31

**Additional context**
It seems that 925acf9eefce1775bb3b174d2742d620243a3412 introduced this regression. @Febbe could you please check where this could come from?

### PR DESCRIPTION
**Pull request by idotobi**
This PR fixes #2539.

Previously only `event.button == 1` was handled in the new input system,
but in the old system all non-zero values where considered.
This lead to drawing events not being handled if a button was pressed
in the new input system.
This is fixed in this commit by handling any `event.button > 0`.

### CODE REVIEW

**Comment by idotobi** [comment]
@rolandlo : I added this to v1.1.0release Milestone as this would be a regression if not fixed before the old input system is turned of.

**Comment by rolandlo** [comment]
> @rolandlo : I added this to v1.1.0release Milestone as this would be a regression if not fixed before the old input system is turned of.
I agree with that. 
Have you tested your PR on Windows as well? Does it change anything with respect to #2628? Otherwise I will make a quick test.

**Comment by Technius** [comment]
Sidenote: these magic constants for button need to be refactored.

**Comment by idotobi** [comment]
> Have you tested your PR on Windows as well? Does it change anything with respect to #2628? Otherwise I will make a quick test.

No I couldn't test it on windows, as I don't have a windows machine :sweat_smile: . If you could do it that would be great.

---

**Regarding #2628**

 I don't think this will fix it completely. I just checked with current master:
If I keep pressing the stylus button that did not work prior to the fix then clicking on anything in the UI of xournalpp did not show any reaction. Hence it could be related.
However, I could still close the app (though that might be special for Linux), and the problem did not persist when I didn't press the Stylus button anymore. So there is probably an additional root cause to this.

You could check if it is linked by putting a debug flag at https://github.com/xournalpp/xournalpp/blob/891c972c27a335113f8a424ae7d3adcbc1ed3f18/src/gui/inputdevices/StylusInputHandler.cpp#L29 or simply printing the `event.button` value to console right before the `if`. If `event.button` gets stuck on a non `1` value than this is part of the cause. In this case, however, we should also find out why it gets stuck as this would cause additional problems.

**Comment by idotobi** [comment]
> Sidenote: these magic constants for button need to be refactored.

Agreed. Best is to create a followup issue for cleaning that up a bit. I can write the issue.
Would you suggest to use enums instead of the magic constants or is there an even better way?

**Comment by rolandlo** [comment]
I have trouble testing the PR on Windows, since currently no stylus buttons work at all (with or without PR). Maybe I have messed up the Windows pen settings.
I have tested the PR on Ubuntu in an 
- X11 session (where my non-eraser stylus button corresponds to the middle mouse button, i.e. GdkEvent Button 2) and in a 
- Wayland session (where my non-eraser stylus button corresponds to the right mouse button, i.e. GdkEvent Button 3). 

When I assign the selection tool to stylus button 2 (i.e. GdkEvent Button 3) in the preferences in a Wayland session, the selection tool will not only be activated if I press the button and touch the screen, but also when I press the button and just hover. This behavior should only occur with `TPCButtonEnabled`.
In conclusion I think this PR would introduce a bug rather than solve a problem.

**Comment by idotobi** [comment]
@rolandlo : You're right :see_no_evil: . I wasn't aware of the TPCButton configuration setting. Luckily @tattsan pointed me to this https://github.com/xournalpp/xournalpp/issues/2539#issuecomment-779197877.

I'll close the PR then. However, I think it might be good to document this both for users and developers. I'll continue the discussion on that in the linked issue.
